custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -99,9 +123,38 @@ def handle_failure(self, response, payload_doc, repeat_record):
         else:
             _save_error_message(payload_doc.domain, payload_doc.case_id, unicode(response.json()))
 
-    def handle_exception(self, exception, repeat_record):
-        if isinstance(exception, RequestConnectionError):
-            _save_error_message(repeat_record.domain, repeat_record.payload_id, unicode(exception))
+
+@RegisterGenerator(NikshayTreatmentOutcomeRepeater, 'case_json', 'JSON', is_default=True)
+class NikshayTreatmentOutcomePayload(BaseNikshayPayloadGenerator):
+
+    def get_payload(self, repeat_record, episode_case):
+        """
+        https://docs.google.com/document/d/1yUWf3ynHRODyVVmMrhv5fDhaK_ufZSY7y0h9ke5rBxU/edit#heading=h.6zwqb0ms7iz9
+        """
+        person_case = get_person_case_from_episode(episode_case.domain, episode_case.get_id)
+        episode_case_properties = episode_case.dynamic_case_properties()
+        base_properties = self._base_properties(repeat_record, person_case)
+        base_properties.update({
+            "PatientID": episode_case_properties.get("nikshay_id"),
+            "OutcomeDate": episode_case_properties.get(TREATMENT_OUTCOME_DATE),
+            "Outcome": treatment_outcome.get(TREATMENT_OUTCOME, '0'),
+            "MO": "{} {}".format(
+                episode_case_properties.get(TREATMENT_SUPPORTER_FIRST_NAME),
+                episode_case_properties.get(TREATMENT_SUPPORTER_LAST_NAME),
```

Comment:
I'm not sure if all this fields a required or not, but if not, we should add default values to these `properties.get()` calls.